File: WinForms/CS/GanttMainDemo/DemosRegistration.cs
using System.Windows.Forms;
using DevExpress.DXperience.Demos;
using DevExpress.DXperience.Demos.CodeDemo;
using DevExpress.XtraEditors;
using DevExpress.XtraTreeList;

namespace DevExpress.XtraGantt.Demos {
    public class DemosInfo : ModulesInfo {
        static TreeList GetTreeListByControl(Control control) {
            if(control == null) return null;
            foreach(Control cntl in control.Controls)
                if(cntl is TreeList)
                    return (TreeList)cntl;
            return null;
        }
        public static void CloseCustomizationForm(Control control) {
            TreeList treeList = GetTreeListByControl(control);
            if(treeList != null)
                treeList.DestroyCustomization();
        }
        //
        public static void ShowModule(string name, GroupControl group, GanttRibbonMenuManager manager) {
            if(Instance != null && Instance.CurrentModuleBase != null) {
                var tControl = Instance.CurrentModuleBase.TModule as TutorialControl;
                if(tControl != null && tControl.IsActiveDemo)
                    return;
            }
            ModuleInfo item = DemosInfo.GetItem(name);
            Cursor currentCursor = Cursor.Current;
            Cursor.Current = Cursors.WaitCursor;
            try {
                Control oldTutorialControl = null;
                if(Instance.CurrentModuleBase != null) {
                    if(Instance.CurrentModuleBase.Name == name)
                        return;
                    oldTutorialControl = Instance.CurrentModuleBase.TModule;
                }
                CloseCustomizationForm(oldTutorialControl);
                CodeTutorialControlBase ctc = item.TModule as CodeTutorialControlBase;
                if(ctc != null)
                    ctc.SelectExampleByName(item.CodeExampleName);
                TutorialControlBase tutorial = item.TModule as TutorialControlBase;
                tutorial.Bounds = group.DisplayRectangle;
                Instance.CurrentModuleBase = item;
                tutorial.Visible = false;
                group.Controls.Add(tutorial);
                tutorial.Dock = DockStyle.Fill;

                TutorialControl tc = item.TModule as TutorialControl;
                if(tc != null) {
                    manager.AllowExport(tc.AllowPrintOptions ? tc.MainControl : null);
                    manager.AllowShowCriticalPath(tc.AllowShowCriticalPath ? tc.MainControl : null);
                    manager.RefreshViewMenu(tc.MainControl);
                } else {
                    manager.RefreshViewMenu(null);
                }

                tutorial.RibbonMenuManager = manager;
                tutorial.TutorialName = name;
                tutorial.Visible = true;
                item.WasShown = true;
                if(oldTutorialControl != null)
                    oldTutorialControl.Visible = false;
                RaiseModuleChanged();
            }
            finally { Cursor.Current = currentCursor; }
        }
    }
}

File: WinForms/VB/GanttMainDemo/DemosRegistration.vb
Imports System.Windows.Forms
Imports DevExpress.DXperience.Demos
Imports DevExpress.DXperience.Demos.CodeDemo
Imports DevExpress.XtraEditors
Imports DevExpress.XtraTreeList

Namespace DevExpress.XtraGantt.Demos

    Public Class DemosInfo
        Inherits ModulesInfo

        Private Shared Function GetTreeListByControl(ByVal control As Control) As TreeList
            If control Is Nothing Then Return Nothing
            For Each cntl As Control In control.Controls
                If TypeOf cntl Is TreeList Then Return CType(cntl, TreeList)
            Next

            Return Nothing
        End Function

        Public Shared Sub CloseCustomizationForm(ByVal control As Control)
            Dim treeList As TreeList = GetTreeListByControl(control)
            If treeList IsNot Nothing Then treeList.DestroyCustomization()
        End Sub

        '
        Public Shared Overloads Sub ShowModule(ByVal name As String, ByVal group As GroupControl, ByVal manager As GanttRibbonMenuManager)
            If Instance IsNot Nothing AndAlso Instance.CurrentModuleBase IsNot Nothing Then
                Dim tControl = TryCast(Instance.CurrentModuleBase.TModule, TutorialControl)
                If tControl IsNot Nothing AndAlso tControl.IsActiveDemo Then Return
            End If

            Dim item As ModuleInfo = GetItem(name)
            Dim currentCursor As Cursor = Cursor.Current
            Cursor.Current = Cursors.WaitCursor
            Try
                Dim oldTutorialControl As Control = Nothing
                If Instance.CurrentModuleBase IsNot Nothing Then
                    If Equals(Instance.CurrentModuleBase.Name, name) Then Return
                    oldTutorialControl = Instance.CurrentModuleBase.TModule
                End If

                CloseCustomizationForm(oldTutorialControl)
                Dim ctc As CodeTutorialControlBase = TryCast(item.TModule, CodeTutorialControlBase)
                If ctc IsNot Nothing Then ctc.SelectExampleByName(item.CodeExampleName)
                Dim tutorial As TutorialControlBase = TryCast(item.TModule, TutorialControlBase)
                tutorial.Bounds = group.DisplayRectangle
                Instance.CurrentModuleBase = item
                tutorial.Visible = False
                group.Controls.Add(tutorial)
                tutorial.Dock = DockStyle.Fill
                Dim tc As TutorialControl = TryCast(item.TModule, TutorialControl)
                If tc IsNot Nothing Then
                    manager.AllowExport(If(tc.AllowPrintOptions, tc.MainControl, Nothing))
                    manager.AllowShowCriticalPath(If(tc.AllowShowCriticalPath, tc.MainControl, Nothing))
                    manager.RefreshViewMenu(tc.MainControl)
                Else
                    manager.RefreshViewMenu(Nothing)
                End If

                tutorial.RibbonMenuManager = manager
                tutorial.TutorialName = name
                tutorial.Visible = True
                item.WasShown = True
                If oldTutorialControl IsNot Nothing Then oldTutorialControl.Visible = False
                Call RaiseModuleChanged()
            Finally
                Cursor.Current = currentCursor
            End Try
        End Sub
    End Class
End Namespace
